frontend/src/pages/NicknamePage/index.tsx

```diff
@@ -35,7 +35,7 @@ const NicknamePage = ({ gameCode }: Props) => {
       // but it does not connect again because playerId and token are falsy
       socket.connect();
     }
-  };
+  }, [browserHistory, gameCode, playerId, token]);
```

Comment:
some of these changes look suspiciously like logic changes that aren't mentioned in the PR description. I assume it's linting related, but what's the rationale for this addition?

Reply:
When I updated the ESLint config for the frontend [eslint-plugin-react-hooks](https://www.npmjs.com/package/eslint-plugin-react-hooks) hadn't been properly configured prior - these changes were just to meet the requirements for the [Rules of Hooks](https://reactjs.org/docs/hooks-rules.html) - so while they are technically logic changes they should only have a positive impact in terms of bugginess.

Reply:
Alright, sounds fair